### ISSUE TITLE
Reflection crash on empty enum in Swift 5.3

### ISSUE THREAD

**Issue by stephencelis** [pointfreeco/swift-case-paths#11]
If you create a case path that goes to `Never` (or any empty enum) using our reflection-based API, any attempted extraction will crash on Swift 5.3 (which is included in Xcode 12 beta):

```swift
enum Foo {
  case bar
  case baz(Never)
}

(/Foo.baz).extract(from: Foo.bar)
```

The issue boils down to this check being wrong:

https://github.com/pointfreeco/swift-case-paths/blob/fb733d87aabb5b053ad782902f2f3d67e0a65ac5/Sources/CasePaths/EnumReflection.swift#L69-L71

Both `enum Foo { case bar }` and `enum Foo {}` have a memory layout of zero. So the unsafe bitcast works for the first but not for the second. In order to distinguish the two we need to look at the type metadata and count the number of cases.

There are a few libraries that accomplish this, including:

- https://github.com/Azoy/Echo
- https://github.com/wickwirew/Runtime

But adding one of these as a dependency is heavyweight, so it'd be better to make the check in a streamlined way. It is relatively straightforward to check if a given type is an enum:

```swift
enum MyType {}

let ptr = unsafeBitCast(MyType.self as Any.Type, to: UnsafeRawPointer.self)
ptr.load(as: Int.self) == 513 // true
```

But unpacking the rest of the work will take some time, so I'm going to put it aside. If anyone is interested in tackling this, please chime in!

**Comment by simme** [comment]
Came here to see if there was an issue for this.

I'm using TCA and noticed that my app would crash inside of CasePaths on any action. Managed to track it down to a pullback on a work in progress view with an uninhabited actions enum. I'm guessing that's related to this issue? Since the crash stopped as soon as I added a case to that empty enum.

**Comment by stephencelis** [comment]
Fixed via #13.
